sklearn/ensemble/_forest.py

```diff
@@ -442,6 +442,10 @@ def fit(self, X, y, sample_weight=None):
         else:
             n_samples_bootstrap = None
 
+        self._n_samples_bootstrap = (
+            n_samples_bootstrap if n_samples_bootstrap is not None else self._n_samples
+        )
```

Comment:
I think we can remove this check from here now and just set this to equal n_samples_bootstrap
```suggestion
        self._n_samples_bootstrap = n_samples_bootstrap
```